### PR TITLE
Constructor injection, @ResponseBody, test

### PR DESCRIPTION
**Pull request by comstogm**

### CODE REVIEW

**Comment by comstogm** [comment]
**Analysis of the program.**
Program looks good, the different layers are present, sites look nice. 

**Was the program available in UC Github on time?**
Yes the program was available right away.

**Is the program easy for you, as an outsider, to understand?**
the program was easy to understand. It was close to the example project, and everything was neatly labelled.

**Does the program compile?**
The program compiled.

**Links to three commits you made to your group's repository during Sprint 1.**
https://github.com/comstogm/GroceryPlanner/commit/3631def06079e47b25e3eecb7312f366a1afb371
https://github.com/comstogm/GroceryPlanner/commit/cef44e13b4b74b1ba0d21de8a02026cbfe5cb4f6
https://github.com/comstogm/GroceryPlanner/commit/cfe1f82186bb65b4bc3d6faec01ada097c840bfa

File: src/main/java/com/adoptmeplus/enterprise/AdoptMePlusController.java [hunk 1]
```diff
@@ -48,6 +52,7 @@ public ResponseEntity fetchAdoptionsById(@PathVariable("id") String id){
      */
 
     @PostMapping(value="/adoptions/create", consumes="application/json", produces="application/json")
+    @ResponseBody
```
Comment (comstogm) [inline src/main/java/com/adoptmeplus/enterprise/AdoptMePlusController.java:55]: https://www.baeldung.com/spring-request-response-body

The @ResponseBody annotation prevents the 500 error when posting a new adoption

File: src/main/java/com/adoptmeplus/enterprise/service/AdoptionServiceStub.java [hunk 2]
```diff
@@ -6,21 +6,17 @@
...
-import com.adoptmeplus.enterprise.dto.Adoption;
-import com.adoptmeplus.enterprise.dto.Dog;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.stereotype.Service;
 import java.util.List;
 
 @Service
 public class AdoptionServiceStub implements IAdoptionService{
 
+    private final IAdoptionDAO adoptionDAO;
```
Comment (comstogm) [inline src/main/java/com/adoptmeplus/enterprise/service/AdoptionServiceStub.java:14]: http://odrotbohm.de/2013/11/why-field-injection-is-evil/
https://www.baeldung.com/constructor-injection-in-spring

Field injection is not preferred.

File: src/test/java/com/adoptmeplus/enterprise/AdoptMePlusApplicationTests.java [hunk 3]
```diff
@@ -49,19 +53,18 @@ void verifyAddAndRemoveAdoptions() {
         adoption.setDogId(DogId);
         adoption.setCustomerId(CustomerId);
 
-        AdoptionServiceStub adoptionService = new AdoptionServiceStub();
         adoptionService.save(adoption);
 
-        List<Dog> adoptions = adoptionService.fetchAll();
```
Comment (comstogm) [inline src/test/java/com/adoptmeplus/enterprise/AdoptMePlusApplicationTests.java:55]: It appears that you were trying to verifyAddAndRemoveAdoptions() but not Dogs.

File: src/test/java/com/adoptmeplus/enterprise/AdoptMePlusApplicationTests.java [hunk 4]
```diff
@@ -49,19 +53,18 @@ void verifyAddAndRemoveAdoptions() {
         adoption.setDogId(DogId);
         adoption.setCustomerId(CustomerId);
 
-        AdoptionServiceStub adoptionService = new AdoptionServiceStub();
```
Comment (comstogm) [inline src/test/java/com/adoptmeplus/enterprise/AdoptMePlusApplicationTests.java:52]: Added service at top

**Review by daltonco — APPROVED** [review]
Approved first 2 changes in AdoptionServiceStub & Controller. Changed CustomerId back to DogId in the third change regarding our AddAndRemove Test

**Pull request merged**